schema/migration.go

```diff
@@ -37,57 +46,365 @@ func (x Versions) Latest() Version { return x[len(x)-1] }
 // VersionDeltaField describes field schema change.
 //
 // If field is deleted then To is equal to `UnknownType`.
-// If field is added then From is equal to `Unknowntype`.
+// If field is added then From is equal to `UnknownType`.
 type VersionDeltaField struct {
 	KeyPath []string  // Key path of the field
 	From    FieldType // The type is changing from this
 	To      FieldType // to this
+
+	// MaxLength int
 }
 
-// VersionDelta contains all fields schema changes in particular schema version.
+// VersionDelta contains all Fields schema changes in particular schema Version.
 type VersionDelta struct {
-	version int32
-	fields  []VersionDeltaField
+	Version int
+	Fields  []*VersionDeltaField
+}
+
+// arrayToMap converts fields array to map, for fast lookups by the field name.
+func arrayToMap(schema []*Field) map[string]*Field {
+	m := make(map[string]*Field)
+
+	for _, v := range schema {
+		m[v.FieldName] = v
+	}
+
+	return m
+}
+
+// buildSchemaDeltaLow is a recursive helper for building schema delta.
+func buildSchemaDeltaLow(keyPath []string, first []*Field, second []*Field) []*VersionDeltaField {
```

Comment:
What does the `Low` part mean in the name? I've seen it in a few function names and I don't know what it means?